lib/api/web-element/scoped-element.js

```diff
@@ -356,6 +359,16 @@ function createNarrowedError({error, condition, timeout}) {
     return err;
   }
 
+  if (error.name === 'InvalidSelectorError') {
+    const err = new Error(
+      `Invalid selector encountered: "${condition}". ` +
+      'Please check the selector syntax and ensure it matches the DOM structure.'
+    );
+    err.name = 'InvalidSelectorError';
+
+    return err;
+  }
+
```

Comment:
Good thinking with this. But we should probably handle this problem of not knowing which particular command is throwing the error for all the errors instead of doing it only for the `InvalidSelectorError`. For that, we can just prepend a simple statement like "Error occurred while trying to locate element "${condition}": " to the `error.message`.

Also, do make sure that `error.message` exists for the error before pretending to it.